doc/design/cli.md

```diff
@@ -0,0 +1,105 @@
+# SQLFlow Command Line Interface
+
+## Overview
+
+At the moment, users can build SQLFlow into a server or a [command line binary](../run/repl.md). Both of the executables have all the functionalities of SQLFlow, and both of them can only run in a Docker container.
+
+For a deployment of SQLFlow in a production environment, the SQLFlow server runs in a Kubernetes cluster. Users access SQLFlow from a Web IDE or the Jupyter Notebook.
+
+For an individual developer, one may deploy SQLFlow on her personal computer as described in [Run SQLFlow Using Docker](../run/docker.md)
+
+For the above two typical scenarios, SQLFlow lacks an easy-to-use user interface for quick start, quick verification, or quick deployment. Jupyter Notebooks and web IDEs still have a certain learning curve that reduces the user experience of SQL programmers. 
+
+From another point of view, a CLI could overtake a GUI in user experience for a simple language as SQLFlow: it's easier to design, easier to implement, easier to obtain. For example, repetitive tasks can be simplified by line editing and history mechanisms. With a CLI program, a user can easily access all the power of SQLFlow from a terminal as:
+
+```bash
+$ sqlflow
+sqlflow> _
+```
+
+To achieve this goal, we also want to hide implementation details about Docker because users don't have to know this. As a result, there are two possible solutions:
+1. Wrap the current SQLFlow REPL in a script, we already have such an [attempt](https://github.com/sql-machine-learning/sqlflow/pull/2114).
+1. Decouple the REPL implementation from the core code of SQLFlow and provide a native binary that only functions as a pure client.
+
+We decide to choose the second solution because it's more lightweight, if a user already has a SQLFlow deployment in her company, all she has to do is to download the native binary.
+
+## Design
+
+### Naming
+
+`REPL` is not a good name, We'll rename the binary as simply `sqlflow`.
+
+### Dependencies and Installation
+
+The original `repl`'s only dependency is the `it2check` bash script that determines whether to call `sixel` to render images.
+
+We'll provide installation scripts on Mac (Bash) and Windows (PowerShell) to install `sqlflow` and its dependencies swiftly. The four scripts are:
```

Comment:
I am afraid that this is too much for users. Let us make sure that users only need to download the client tool before running it. If necessary, let us abandon it2check and other cool stuff at the right moment.

Reply:
Done. 
The client will only provide auto-complete features for built-in models (canned estimators and XGBoost) at the 1st stage. 
The auto-complete features for `sqlflow_models` and Model Zoo models will be designed at the next state when necessary.

`it2check` has nothing to do with auto-completion, we'll avoid this dependency at the first stage